util/src/builder/select_menu_option.rs

```diff
@@ -0,0 +1,555 @@
+    /// Label is too long.
+    LabelTooLong {
+        /// Provided label.
+        label: String,
+    },
+    /// Value is empty.
+    ValueEmpty {
+        /// Provided value although it's empty.
+        value: String,
+    },
+    /// Value is too long.
+    ValueTooLong {
+        /// Provided value.
+        value: String,
+    },
+    /// Description is too long.
+    DescriptionTooLong {
+        /// Provided description.
+        description: String,
+    },
+}
+
+/// Create a [`SelectMenuOption`] with a builder.
+///
+/// # Example
+/// ```
+/// use twilight_util::builder::{select_menu_option::SelectMenuOptionBuilder, CallbackDataBuilder};
+/// use twilight_model::{
+///     channel::message::MessageFlags,
+///     application::component::{Component, SelectMenu},
+///     channel::ReactionType,
+///     id::EmojiId,
+/// };
+///
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let component = Component::SelectMenu(
+///     SelectMenu {
+///         custom_id: "characters".into(),
+///         disabled: false,
+///         max_values: None,
+///         min_values: None,
+///         options: vec![
+///             SelectMenuOptionBuilder::new("Twilight Sparkle".to_string(), "twilight-sparkle".to_string())
+///                 .default(true)
+///                 .emoji(ReactionType::Custom {
+///                     animated: false,
+///                     id: EmojiId::new(754728776402993173_u64).unwrap(),
+///                     name: Some("sparkle".to_string()),
+///                 })
+///                 .build()?,
+///             SelectMenuOptionBuilder::new("Rarity".to_string(), "rarity".to_string())
+///                 .emoji(ReactionType::Custom {
+///                     animated: false,
+///                     id: EmojiId::new(765306914153299978_u64).unwrap(),
+///                     name: Some("rarsmile".to_string()),
+///                 })
+///                 .build()?,
+///         ],
+///         placeholder: None,
+///     }
+/// );
+///
+/// let callback_data = CallbackDataBuilder::new()
+///     .content("Callback message".to_string())
+///     .flags(MessageFlags::EPHEMERAL)
+///     .components([component.clone()])
+///     .build();
+///
+/// assert_eq!(callback_data.components, Some(vec![component]));
+/// # Ok(()) }
+/// ```
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[must_use = "builders have no effect if unused"]
+pub struct SelectMenuOptionBuilder(SelectMenuOption);
+
+impl SelectMenuOptionBuilder {
+    /// The maximum amount of characters which a label can have.
+    pub const LABEL_LENGTH_LIMIT: usize = 100;
+
+    /// The maximum amount of characters which a value can have.
+    pub const VALUE_LENGTH_LIMIT: usize = 100;
+
+    /// The maximum amount of characters which a description can have.
+    pub const DESCRIPTION_LENGTH_LIMIT: usize = 100;
+
+    /// Create a new builder to construct a [`SelectMenuOption`].
+    pub const fn new(label: String, value: String) -> Self {
```

Comment:
I think these two parameters should be reversed, as `value` is the unique option identifier.

```suggestion
    pub const fn new(value: String, label: String) -> Self {
```